scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -52,6 +87,18 @@ trait ExecutionContext {
       }
       val flow = mode.newFlowConnector(withId).connect(flowDef)
 
+      flow match {
+        case hadoopFlow: HadoopFlow =>
+          val flowSteps = hadoopFlow.getFlowSteps.asScala
+          flowSteps.foreach(step => {
+            val baseFlowStep: BaseFlowStep[JobConf] = step.asInstanceOf[BaseFlowStep[JobConf]]
+            val descriptions = getDesc(baseFlowStep)
+            updateFlowStepName(baseFlowStep, descriptions)
```

Comment:
The mapred.name is already set by the time it gets here, so the job.xml file for hadoop will not have any mention of the descriptions. However, the "stepName" needs to be updated for the dot file generation to have the descriptions